contracts/src/v0.8/dev/AuthorizedReceiver.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./interfaces/AuthorizedReceiverInterface.sol";
+
+abstract contract AuthorizedReceiver is AuthorizedReceiverInterface {
+  mapping(address => bool) private s_authorizedSenders;
+  address[] private s_authorizedSenderList;
+
+  event AuthorizedSendersChanged(address[] senders, address changedBy);
+
+  /**
+   * @notice Sets the fulfillment permission for a given node. Use `true` to allow, `false` to disallow.
+   * @param senders The addresses of the authorized Chainlink node
+   */
+  function setAuthorizedSenders(address[] calldata senders) external override validateAuthorizedSenderSetter {
+    require(senders.length > 0, "Must have at least 1 authorized sender");
+    // Set previous authorized senders to false
+    uint256 authorizedSendersLength = s_authorizedSenderList.length;
+    for (uint256 i = 0; i < authorizedSendersLength; i++) {
+      s_authorizedSenders[s_authorizedSenderList[i]] = false;
+    }
+    // Set new to true
+    for (uint256 i = 0; i < senders.length; i++) {
+      s_authorizedSenders[senders[i]] = true;
+    }
```

Comment:
Why not delete the keys that are not in <senders> and then add those that don't exist in <s_authorizedSenders> yet?